buildSrc/subprojects/ide/src/main/kotlin/org/gradle/gradlebuild/ide/IdePlugin.kt

```diff
@@ -77,230 +56,120 @@ limitations under the License."""
 open class IdePlugin : Plugin<Project> {
 
     override fun apply(project: Project): Unit = project.run {
-        configureIdeaForRootProject()
+        if (providers.environmentVariable("idea.active").getOrElse("false").toBoolean()) {
```

Comment:
I thought this is a system property and not an environment variable. Also, can we be sure that we only should apply the idea plugin when we run things from IDEA? What does that mean for caching, aka will builds from IDEA always have a different classloader since a new plugin is present?

Reply:
🤦 Sorry, about " system property". Thanks for noticing.

I added this coming from this: https://youtrack.jetbrains.com/issue/IDEA-239754

It does not work anymore (I'll comment on the IDEA issue above why), but if you request the IdeaModel, we automatically apply the idea plugin:
https://github.com/gradle/gradle/blob/cb3b52cd75176e51ab7e75a5ba7ccd992fb9a271/subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/IdeaModelBuilder.java#L82

 So I thought this would be the idiomatic way to do this (ideally I would like to do `withPlugin(IdeaPlugin)` instead of using a system property.

But given this comment

> What does that mean for caching, aka will builds from IDEA always have a different classloader since a new plugin is present?

I wonder if it is actually a good practice. Also of Gradle itself to automatically apply an additional plugin if a certain model is request.

Maybe I should not do this change and always apply the plugin? Wdyt?

Reply:
Either always apply the plugin or at least add the `idea-ext` plugin to the root project classpath without applying (`apply = false`).

We should check what the performance impact is, and if it is negligible, then we should always apply the plugin.

Reply:
I am sure the performance impact is negligible. Goal is to have a readable build that can act a a good example for others.

Lets always apply the plugin (as we did before).